src/ring_theory/polynomial/cyclotomic.lean

```diff
@@ -517,6 +518,84 @@ begin
   exact neg_inj.mp (eq.symm heq)
 end
 
+/-- If `(a : ℕ)` is a root of `cyclotomic n (zmod p)`, where `p` is a prime, then `a` and `p` are
+coprime. -/
+lemma coprime_of_root_cyclotomic {n : ℕ} (hpos : 0 < n) {p : ℕ} [hprime : fact p.prime] {a : ℕ}
+  (hroot : is_root (cyclotomic n (zmod p)) (nat.cast_ring_hom (zmod p) a)) :
+  a.coprime p :=
+begin
+  apply nat.coprime.symm,
+  rw [nat.prime.coprime_iff_not_dvd hprime],
+  by_contra h,
+  replace h := (zmod.nat_coe_zmod_eq_zero_iff_dvd a p).2 h,
+  rw [is_root.def, ring_hom.eq_nat_cast, h, ← coeff_zero_eq_eval_zero] at hroot,
+  by_cases hone : n = 1,
+  { simp only [hone, cyclotomic_one, zero_sub, coeff_one_zero, coeff_X_zero, neg_eq_zero,
+    one_ne_zero, coeff_sub] at hroot,
+    exact hroot },
+  rw [cyclotomic_coeff_zero (zmod p) (nat.succ_le_of_lt (lt_of_le_of_ne
+    (nat.succ_le_of_lt hpos) (ne.symm hone)))] at hroot,
+  exact one_ne_zero hroot
+end
+
 end cyclotomic
 
+section order
+
+/-- If `(a : ℕ)` is a root of `cyclotomic n (zmod p)`, where `p` is a prime that does not divide
+`n`, then the multiplicative order of `a` modulo `p` is exactly `n`. -/
+lemma order_of_root_cyclotomic {n : ℕ} (hpos : 0 < n) {p : ℕ} [hprime : fact p.prime] {a : ℕ}
+  (hn : ¬ p ∣ n) (hroot : is_root (cyclotomic n (zmod p)) (nat.cast_ring_hom (zmod p) a)) :
+  order_of (zmod.unit_of_coprime a (coprime_of_root_cyclotomic hpos hroot)) = n :=
+begin
+  have ha := coprime_of_root_cyclotomic hpos hroot,
+  have hord : (zmod.unit_of_coprime a ha) ^ n = 1,
```

Comment:
The proof is somewhat longish. Do you think you can factor out some reusable sublemmas? Maybe this `have` statement? Maybe some other parts?

Reply:
I've put that the order divides `n` in a separate lemma, that is a reasonably interesting result in its own, and also that if `m` is a proper divisor of `n`, then `X ^ m - 1` divides `∏ i in nat.proper_divisors n, cyclotomic i R`.

The proof is now slightly shorter, I do not see any other interesting lemmas to factor out.